abcde does not work on single track CD

# THE PROBLEM 

abcde-musicbrainz-tool is failing on a CD with only one track on it.  It is expecting 5 or more arguments in the `--discinfo` arguments, but when you have a CD with only one track on it, only 4 arguments are passed.  There is a guard for this as well.

# TO REPRODUCE THE PROBLEM

For example, here is how abcde-musicbrainz-tool gets invoked on a CD with one track:
```
$ abcde-musicbrainz-tool --command calcid --discinfo 1 1 150 106366 0 
[ERROR] abcde-musicbrainz-tool Insufficient or missing discinfo data.
```

# EXPLORATORY HACKING

When I tack on an extra 0, it tricks it into working:
```
:~/Music$ abcde-musicbrainz-tool --command calcid --discinfo 1 1 150 106366 0 0
8umh4uW3lys1qylvDBm_bz5AK1A-
```

Here is an example with a CD with more than 1 track (4 in this case):
```
$ abcde-musicbrainz-tool --command calcid --discinfo 1 4  150 237259 0 68226 105606 152018
JtPoRT9T6A7_MZm1u4afgWcTpGk-
```
# THE BUG

abcde-musicbrain-tool's GetOptions is set up to expect 5 or more discinfo arguments:
```
GetOptions ("device=s"       => \$device,
            "command=s"      => \$command,
            "discid=s"       => \$discid,
            "discinfo=i{5,}" => \@discinfo,
            "workdir=s"      => \$workdir,
            "start=s"        => \$start,
            "help|h"         => \$help,
            "man"            => \$man) or pod2usage(-verbose => 0, -exitcode => 2);
```

There is also a guard that exits the program if there are less than 5 discinfo arguments:
```
    if ($#discinfo < 5) {
	print STDERR "Insufficient or missing discinfo data.\n";
	exit(1);
    }
```

# THE SOLUTION

Change both those 5's to 4's.

```
GetOptions ("device=s"       => \$device,
            "command=s"      => \$command,
            "discid=s"       => \$discid,
            "discinfo=i{4,}" => \@discinfo,
            "workdir=s"      => \$workdir,
            "start=s"        => \$start,
            "help|h"         => \$help,
            "man"            => \$man) or pod2usage(-verbose => 0, -exitcode => 2);
```

There is also a guard that exits the program if there are less than 5 discinfo arguments:
```
    if ($#discinfo < 4) {
	print STDERR "Insufficient or missing discinfo data.\n";
	exit(1);
    }
```

```
$ git diff
diff --git a/abcde-musicbrainz-tool b/abcde-musicbrainz-tool
index 51e49c2..3acbc0e 100755
--- a/abcde-musicbrainz-tool
+++ b/abcde-musicbrainz-tool
@@ -28,7 +28,7 @@ Getopt::Long::Configure ('no_auto_abbrev');
 GetOptions ("device=s"       => \$device,
             "command=s"      => \$command,
             "discid=s"       => \$discid,
-            "discinfo=i{5,}" => \@discinfo,
+            "discinfo=i{4,}" => \@discinfo,
             "workdir=s"      => \$workdir,
             "start=s"        => \$start,
             "help|h"         => \$help,
@@ -302,7 +302,7 @@ if ($command =~ m/^id/) {
     # Calculate MusicBrainz ID from disc offsets; see
     # https://musicbrainz.org/doc/DiscIDCalculation
 
-    if ($#discinfo < 5) {
+    if ($#discinfo < 4) {
        print STDERR "Insufficient or missing discinfo data.\n";
        exit(1);
     }
```

# REFERENCES

Here is a bug report for this problem - https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=907852

The Perl GetOpt::Long documentation explains the "n or more" syntax (https://perldoc.perl.org/Getopt::Long):
```
Options can take multiple values at once, ...

This can be accomplished by adding a repeat specifier to the option specification. Repeat specifiers are very similar to the {...} repeat specifiers that can be used with regular expression patterns. ...

The destination for the option must be an array or array reference.

It is also possible to specify the minimal and maximal number of arguments an option takes. foo=s{2,4} indicates an option that takes at least two and at most 4 arguments. foo=s{1,} indicates one or more values; foo:s{,} indicates zero or more option values.
```

--- a/abcde-musicbrainz-tool
+++ b/abcde-musicbrainz-tool
@@ -28,7 +28,7 @@ Getopt::Long::Configure ('no_auto_abbrev');
 GetOptions ("device=s"       => \$device,
             "command=s"      => \$command,
             "discid=s"       => \$discid,
-            "discinfo=i{5,}" => \@discinfo,
+            "discinfo=i{4,}" => \@discinfo,
             "workdir=s"      => \$workdir,
             "start=s"        => \$start,
             "help|h"         => \$help,
@@ -302,7 +302,7 @@ if ($command =~ m/^id/) {
     # Calculate MusicBrainz ID from disc offsets; see
     # https://musicbrainz.org/doc/DiscIDCalculation
 
-    if ($#discinfo < 5) {
+    if ($#discinfo < 4) {
 	print STDERR "Insufficient or missing discinfo data.\n";
 	exit(1);
     }

